components/src/preact/dateRangeSelector/date-range-selector.stories.tsx

```diff
@@ -134,37 +125,83 @@ export const SetCorrectInitialDateTo: StoryObj<DateRangeSelectorProps> = {
     play: async ({ canvasElement }) => {
         const canvas = within(canvasElement);
 
-        const dateFrom = () => canvas.getByPlaceholderText('Date from');
-        const dateTo = () => canvas.getByPlaceholderText('Date to');
-        const selectField = () => canvas.getByRole('combobox');
-
         await waitFor(() => {
-            expect(selectField()).toHaveValue('Custom');
-            expect(dateFrom()).toHaveValue(earliestDate);
-            expect(dateTo()).toHaveValue(initialDateTo);
+            expect(selectField(canvas)).toHaveValue('Custom');
+            expect(dateFromPicker(canvas)).toHaveValue(earliestDate);
+            expect(dateToPicker(canvas)).toHaveValue(initialDateTo);
         });
     },
 };
 
 export const ChangingDateSetsOptionToCustom: StoryObj<DateRangeSelectorProps> = {
     ...Primary,
-    play: async ({ canvasElement }) => {
-        const canvas = within(canvasElement);
-
-        const dateFrom = () => canvas.getByPlaceholderText('Date from');
-        const dateTo = () => canvas.getByPlaceholderText('Date to');
-        const selectField = () => canvas.getByRole('combobox');
+    play: async ({ canvasElement, step }) => {
+        const { canvas, filterChangedListenerMock, optionChangedListenerMock } = await prepare(canvasElement, step);
 
         await waitFor(() => {
-            expect(selectField()).toHaveValue('Last month');
+            expect(selectField(canvas)).toHaveValue('Last month');
         });
 
-        await userEvent.type(dateFrom(), '{backspace>12}');
-        await userEvent.type(dateFrom(), '2000-01-01');
-        await userEvent.click(dateTo());
+        step('Change date to custom value', async () => {
+            await userEvent.type(dateFromPicker(canvas), '{backspace>12}');
+            await userEvent.type(dateFromPicker(canvas), '2000-01-01');
+            await userEvent.click(dateToPicker(canvas));
+
+            await waitFor(async () => {
+                expect(selectField(canvas)).toHaveValue('Custom');
+            });
```

Comment:
You either have to `await` the `expect` or you remove the curly braces (making the `expect` the return value)
```suggestion
            await waitFor(async () => 
                expect(selectField(canvas)).toHaveValue('Custom')
            );
```